google-ads-examples/src/main/java/com/google/ads/googleads/examples/reporting/ParallelReportDownload.java

```diff
@@ -0,0 +1,253 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.ads.googleads.examples.reporting;
+
+import com.beust.jcommander.Parameter;
+import com.google.ads.googleads.examples.utils.ArgumentNames;
+import com.google.ads.googleads.examples.utils.CodeSampleParams;
+import com.google.ads.googleads.lib.GoogleAdsClient;
+import com.google.ads.googleads.v3.errors.GoogleAdsError;
+import com.google.ads.googleads.v3.errors.GoogleAdsException;
+import com.google.ads.googleads.v3.services.GoogleAdsServiceClient;
+import com.google.ads.googleads.v3.services.SearchGoogleAdsStreamRequest;
+import com.google.ads.googleads.v3.services.SearchGoogleAdsStreamResponse;
+import com.google.api.gax.rpc.ResponseObserver;
+import com.google.api.gax.rpc.StreamController;
+import com.google.common.collect.ImmutableList;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.SettableFuture;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class ParallelReportDownload {
+
+  // Adjust these fields as required.
+  /** Defines the Google Ads Query Language (GAQL) query strings to run for each customer ID. */
+  private static final List<String> GAQL_QUERY_STRINGS =
+      ImmutableList.of(
+          "SELECT campaign.id, metrics.impressions, metrics.clicks"
+              + " FROM campaign"
+              + " WHERE segments.date DURING LAST_30_DAYS",
+          "SELECT campaign.id, ad_group.id, metrics.impressions, metrics.clicks"
+              + " FROM ad_group"
+              + " WHERE segments.date DURING LAST_30_DAYS");
+
+  /** Specifies the maximum duration that this code example may run for. */
+  private static final int MAX_WAIT_TIME_SECONDS = 60 * 60; // 1 hour.
+
+  /** Absolute wall clock time by when the job must be completed. */
+  private static final long DEADLINE_MILLIS =
+      System.currentTimeMillis() + MAX_WAIT_TIME_SECONDS * 1000;
+
+  private static class ParallelReportDownloadParams extends CodeSampleParams {
+
+    @Parameter(names = ArgumentNames.CUSTOMER_ID, required = true)
```

Comment:
Consider adding a `description` attribute for this annotation since it's a list, and 99% of our other examples have `CUSTOMER_ID` as a single value.